src/posts/2021-06-07-fixing-contrast-issues-on-your-own-site-and-elsewhere.md

```diff
@@ -0,0 +1,94 @@
+
+For the most part, text contrast issues can be automatically found, and as a developer whenever something can be automated I prefer doing that over repeating an action (like selecting elements or clicking around with a color picker) over and over again. The chance of me missing something increases the more I have to repeat myself.
+
+Instead of doing that manual work I wrote a color contrast checker (part of the web browser for developers I build, Polypane) that automatically goes over all text elements, resolves their font size, color and (actual) background and uses that to calculate the contrast ratio.
+
+Still though, that just helps you find inaccessible colors slightly faster. What now? You could use a color picker to pick a darker (or lighter) color and try again, and maybe you have to do that a few times before you get it right. You could push the problem back to your designer and ask them for a new color, or if you have a design system, switch to the next hue in the palette. All of these take you outside of the flow you're currently in.
+
+You can also calculate the nearest color that does have enough contrast. So that's what Polypane does: when it finds a contrast issue, it will suggest the nearest sufficiently contrasting color for you to use. You can copy it or preview it, all without getting out of the flow you're currently in.
+
+Here's what that looks like:
+
+<video src="/img/posts/2021-06-07-fixing-contrast-issues-on-your-own-site-and-elsewhere/copy-color.mp4" loop="" controls="">Example of a contrast label in Polypane, which shows a failing score, a suggested color that gets applied when you hover over it, and copies that color when you click it.</video>
+
+This is a really useful feature that I didn't wanna keep it locked away in my own product, so it's also available online. When you enter two colors here, you'll get suggestions of similar colors with enough contrast: [color contrast checker](https://polypane.app/color-contrast/).
+
+As an author, it is our responsibility to fix these contrast issues and luckily, computers can help us with this a whole lot.
+
+## But what if you're not the author?
+
+We can fix our own sites, but what about _other people's websites_? After all, the chance you're visiting a website that has low contrast text is nearly nine out of ten times.
+
+I realized I could use the code I wrote to automatically fix websites instead. It doesn't make sense inside Polypane, but it would be really useful as a browser extension.
+
+That browser extension is called [Fix Contrast](https://fixa11y.com) and it freely available for Chrome, Edge, Firefox, Vivaldi and Brave, and hopefully soon for Opera and Safari too.
+
+After you install it, it will automatically fix any text with low contrast up to your preferred contrast ratio.
+
+![The Fix Contrast settings](/img/posts/2021-06-07-fixing-contrast-issues-on-your-own-site-and-elsewhere/ui.png)
```

Comment:
Can we be a little more descriptive of what the settings contains?

Reply:
Happy to, but because I already describe that in the paragraph below, that might be a little too much of a good thing? In any case that's why I kept the alt a little more genetic.